### DISCUSSION TITLE
Props in QCheckbox for custom icons

### DISCUSSION THREAD

**Discussion by dsl101** [quasarframework/quasar#12232] [Ideas / Proposals] [answered]
I have a need to create a set of controls that look and behave in a very similar way, but need custom icons for checkbox-type controls, which will sit right next to a regular QCheckbox. Emulating the look and behaviour of QCheckbox from scratch isn't trivial, and it would be ideal if I were able to do something like:
```
<q-checkbox v-model="starred" label="Show starred items" icon-true="star" icon-false="star_border" />
```
I think, apart from the additional props (with perhaps `icon-indet` for completeness), it would only require changes in QCheckbox.js in this section of code:
```
const bgNode = h('div', {
  class: 'q-checkbox__bg absolute'
}, [
  h('svg', {
    class: 'q-checkbox__svg fit absolute-full',
    viewBox: '0 0 24 24',
    'aria-hidden': 'true'
  }, [
    h('path', {
      class: 'q-checkbox__truthy',
      fill: 'none',
      d: 'M1.73,12.91 8.1,19.28 22.79,4.59'
    }),

    h('path', {
      class: 'q-checkbox__indet',
      d: 'M4,14H20V10H4'
    })
  ])
])
```

Alternatives I've considered are building all the components from scratch using QItem or QBtn, but the behaviour and look of QCheckbox is really what I'm after.

**Comment by dsl101** [comment]
I tried hacking this into QCheckbox, and it kind of demonstrates what I'm looking for, without any of the props (it just changes all checkboxes to stars)
```
        h('i', {
          staticClass: 'q-icon notranslate material-icons fit absolute-full'
        }, [
          h('span', { staticClass: this.value ? 'q-checkbox__truthy' : undefined }, this.value ? 'star' : 'star_border'),
        ] )
```

![image](https://user-images.githubusercontent.com/900542/151530615-e7e851cd-6577-4bbe-81f4-93c9a6b34d31.png)

The sizing is wrong of course too, as I guess the svg takes care of that for regular checkboxes.

**Comment by rstoenescu** [comment, marked as answer]
Implementing checked-icon, unchecked-icon, indeterminate-icon props for QCheckbox.
Follow: https://github.com/quasarframework/quasar/issues/12233

**Reply by rstoenescu** [reply]
QCheckbox & QRadio with custom icons will be available in Quasar v2.5.0.

**Reply by dsl101** [reply]
That's fantastic—many thanks for that. Any chance of a v1 backport? Unfortunately this is on a legacy project :(

**Reply by rstoenescu** [reply]
It's going to be available in v1.18 as well.
